### PR TITLE
Add review react page

### PR DESCRIPTION
**Pull request by nirjacob**
*added AddReview react page
*tweaked general design

### CODE REVIEW

File: controllers/podcast.js [hunk 1]
```diff
@@ -1,86 +1,86 @@
...
-        (err) {
-        return next(err)
-    }
+  const numberOfPodcasts = parseInt(req.params.number)
+  try {
+    const podcastsInfo = await getBestPodcasts(numberOfPodcasts)
+    if (!podcastsInfo) return res.status(404).send('No podcasts to present')
+    return res.status(200).send(podcastsInfo)
+  } catch
+    (err) {
```
Comment (racheli-lavan) [inline controllers/podcast.js:17]: indentation...why is the (err) in a new line?

File: controllers/podcast.js [hunk 2]
```diff
@@ -1,86 +1,86 @@
...
-        (err) {
-        return next(err)
-    }
+  const queryParams = req.params.query
+  try {
+    const podcastsInfo = await getPodcastSearchResults(queryParams)
+    if (!podcastsInfo) return res.status(404).send('Podcasts not found')
+    return res.status(200).send(podcastsInfo)
+  } catch
+    (err) {
```
Comment (racheli-lavan) [inline controllers/podcast.js:28]: same here

File: controllers/podcast.js [hunk 3]
```diff
@@ -1,86 +1,86 @@
...
-    } catch
-        (err) {
-        return next(err)
-    }
+  try {
+    const newPodcast = req.body
+    await savePodcastToDb(newPodcast)
+    return res.status(200).send('Podcasts has been successfully added')
+  } catch
+    (err) {
```
Comment (racheli-lavan) [inline controllers/podcast.js:49]: same here

File: react-client/src/containers/AddReview/AddReview.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      return selectedRating < (index + 1) ? '\u2605' : '\u2606'
+    })
+    setRatingStars(newRate)
+  }
+
+  async function ratePodcast() {
+    const review = {
+      podcastId: podcastId,
+      text: text,
+      rating: 10 - rating
```
Comment (racheli-lavan) [inline react-client/src/containers/AddReview/AddReview.js:27]: why 10 - rating?

File: react-client/src/containers/AddReview/AddReview.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import style from './AddReview.module.scss'
+import { useState } from 'react'
+import { Link, useLocation } from 'react-router-dom'
+import { addNewReview } from '../../services/Podcasts'
+
+const AddReview = () => {
+  const location = useLocation()
+  const podcastId = location.pathname.replace('/podcast/new-review/', '')
+  const [ratingStars, setRatingStars] = useState(['\u2606', '\u2606', '\u2606', '\u2606', '\u2606'])
```
Comment (racheli-lavan) [inline react-client/src/containers/AddReview/AddReview.js:10]: the state of the rating should be a number and not a font. in the ui you should translate this number into a font

File: react-client/src/containers/EditPodcast/EditPodcast.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+const EditPodcast = () => {
+  const location = useLocation()
+  const podcastId = location.pathname.replace('/podcast/edit/', '')
+  const [podcast, setPodcast] = useState({})
+
+  async function deleteEditedPodcast() {
+    try {
+      await deletePodcast(podcastId)
+    } catch (err) {
+      console.error(`${err}`)
```
Comment (racheli-lavan) [inline react-client/src/containers/EditPodcast/EditPodcast.js:16]: and how will the user know that the deletion didn't work?

File: react-client/src/containers/EditPodcast/EditPodcast.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  }
+
+  async function updateEditedPodcast() {
+    try {
+      await updatePodcast(podcastId, podcast)
+    } catch (err) {
+      console.error(`${err}`)
```
Comment (racheli-lavan) [inline react-client/src/containers/EditPodcast/EditPodcast.js:24]: same here - how will the user know that the save didn't work?

File: react-client/src/containers/EditPodcast/EditPodcast.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  }
+
+  async function setData() {
+    try {
+      const podcastById = await getPodcast(podcastId)
+      setPodcast({
```
Comment (racheli-lavan) [inline react-client/src/containers/EditPodcast/EditPodcast.js:31]: why not just do setPodcast(podcastById) ?

File: controllers/podcast.js (hunk 1 above)
Comment (nirjacob) [reply controllers/podcast.js:17]: eslint at it again...

**Review by racheli-lavan — APPROVED** [review]

**Pull request merged**